File: functions/alsa-lib/variants/go/2_muldiv_down_6.go
package main

func muldivDown(a, b, c int32) int32 {
	v := (int64(a) * int64(b)) / int64(c)
	switch {
	case v > 2147483647:
		return 2147483647
	case v < -2147483648:
		return -2147483648
	default:
		return int32(v)
	}
}


func main(){
    muldivDown(0,0,0)
}

File: functions/alsa-lib/variants/c/2_muldiv_down_6.c
#include <stdint.h>

static inline int muldiv_down(int a, int b, int c)
{
	int64_t v = (int64_t)a * b / c;
	if (v > 2147483647LL) {
		v = 2147483647;
	} else if (v < -2147483648LL) {
		v = -2147483648;
	}
	return (int)v;
}


int main(){
    muldiv_down(0,0,0);
}
